src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -152,6 +152,14 @@ public function start()
             throw new \RuntimeException(sprintf('Failed to start the session because headers have already been sent by "%s" at line %d.', $file, $line));
         }
 
+        $sessionName = session_name();
+        $sessionId = $_COOKIE[$sessionName] ?? null;
+        // validate according to https://www.php.net/manual/fr/function.session-start.php
+        if ($sessionId && !preg_match('/^[a-zA-Z0-9,\-]{1,123}$/', $sessionId)) {
```

Comment:
according to https://stackoverflow.com/questions/12240922/what-is-the-length-of-a-php-session-id-string, we'd rather go with `'/^[a-zA-Z0-9,-]{22,}$/'` for the regexp.